lib/asset/manifest/src/Obelisk/Asset/TH.hs

```diff
@@ -32,20 +32,45 @@ assetPath root relativePath =
 staticPrefix :: FilePath
 staticPrefix = "/static"
 
+-- | Location of the symbolic link to static assets and resources.
+--
+-- We shouldn't be defining it here, but apparently this is just hard-coded where needed.
+staticOutPath :: FilePath
+staticOutPath = "static.out"
+
+-- | Embed a filepath via template haskell. Resources embedded this way
+-- are requested from the "/static" route.
+--
+-- If the filepath can not be found in the static output directory,
+-- this will throw a compile-time error.
 staticAssetRaw :: FilePath -> Q Exp
-staticAssetRaw fp = do
-  exists <- runIO $ doesFileExist $ "static.out/" <> fp
-  when (not exists) $
-    fail $ "The file " <> fp <> " was not found in static.out"
-  returnQ $ LitE $ StringL $ staticPrefix </> fp
+staticAssetRaw = staticAssetWorker staticPrefix staticOutPath
 
 staticAssetHashed :: FilePath -> FilePath -> Q Exp
 staticAssetHashed root fp = do
   LitE . StringL . (staticPrefix </>) <$> hashedAssetFilePath root fp
 
+-- | Embed a filepath via template haskell. Differently to 'staticAssetRaw'
+-- this points to a local filepath instead of an URL during deployment.
+--
+-- If the filepath can not be found in the static output directory,
+-- this will throw a compile-time error.
 staticAssetFilePathRaw :: FilePath -> FilePath -> Q Exp
-staticAssetFilePathRaw root fp = returnQ $ LitE $ StringL $ root </> fp
+staticAssetFilePathRaw root = staticAssetWorker root staticOutPath
 
 staticAssetFilePath :: FilePath -> FilePath -> Q Exp
 staticAssetFilePath root fp = do
   LitE . StringL . (root </>) <$> hashedAssetFilePath root fp
+
+-- | @'staticAssetWorker' root staticOut fp@.
+--
+-- Produces @root </> fp@, but checks before that @fp@ has been copied
+-- to 'staticOutPath' and produces a compilation error otherwise.
+-- This helps finding typos in filepaths, etc... at compile-time instead of
+-- run-time.
+staticAssetWorker :: FilePath -> FilePath -> FilePath -> Q Exp
```

Comment:
I know the arguments are explained in the top-level haddock, but it would be nice to have docs on the arguments themselves too